src/channel/queuedChannel.ts

```diff
@@ -29,19 +29,19 @@ export class QueuedChannel<T> implements OutputChannel<T> {
 
     public publish(message: T): Promise<void> {
         if (this.closed) {
-            return Promise.reject(new Error('Channel is closed.'));
+            return Promise.reject(new MessageDeliveryError('Channel is closed.', false));
```

Comment:
Boolean as a positional argument is hard to read and reason about. What about making it like this:
```suggestion
            return Promise.reject(new MessageDeliveryError('Channel is closed.', {retryable:false}));
```

Reply:
I have the same opinion. But since we have no plans to support other properties, I would go with named constructors. Wdt?

Reply:
Sounds great!